Source/DafnyLanguageServer.Test/GutterStatus/ConcurrentLinearVerificationGutterStatusTester.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Microsoft.Dafny.LanguageServer.IntegrationTest.Extensions;
+using Microsoft.Dafny.LanguageServer.IntegrationTest.Util;
+using Microsoft.Dafny.LanguageServer.Workspace.Notifications;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using OmniSharp.Extensions.JsonRpc;
+using OmniSharp.Extensions.LanguageServer.Protocol.Document;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+using Range = OmniSharp.Extensions.LanguageServer.Protocol.Models.Range;
+
+namespace Microsoft.Dafny.LanguageServer.IntegrationTest.Diagnostics;
+
+[TestClass]
+public class ConcurrentLinearVerificationGutterStatusTester : LinearVerificationGutterStatusTester {
+  private const int MaxSimultaneousVerificationTasks = 3;
+
+  protected TestNotificationReceiver<VerificationStatusGutter>[] verificationStatusGutterReceivers =
+    new TestNotificationReceiver<VerificationStatusGutter>[MaxSimultaneousVerificationTasks];
+
+  private void NotifyAllVerificationGutterStatusReceivers(VerificationStatusGutter request) {
+    foreach (var receiver in verificationStatusGutterReceivers) {
+      receiver.NotificationReceived(request);
```

Comment:
I'm confused by this. All verificationStatusGutterReceivers receive the same notifications, which would be the notifications for all the tasks. How does `VerifyTrace` that gets `verificationStatusGutterReceivers[i]` not get confused by the bogus notifications?

Reply:
It checks that the filename matches and filters out notifications that do not match.
That way, it can rebuild the trace for every file independently.

Reply:
Added a commit in https://github.com/dafny-lang/dafny/pull/1946/ to add a comment regarding this.